### PR TITLE
Add zenodo badge

### PR DESCRIPTION
**Pull request by sklam**
Closes https://github.com/numba/numba/issues/6572

as titled

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -10,6 +10,10 @@ Numba
    :target: https://numba.discourse.group/
    :alt: Discourse
 
+.. image:: https://zenodo.org/badge/3659275.svg
+   :target: https://zenodo.org/badge/latestdoi/3659275
+   :alt: Zenodo
```
Comment (stuartarchibald) [inline README.rst:15]:
```suggestion
   :alt: Zenodo DOI
```

**Comment by esc** [comment]
It looks good. What did you have to do on Zenodo side, did it require making an account? Also, where was the list of authors source from, is it automatically derived from git history?

**Review by stuartarchibald — APPROVED** [review]
Thanks for the fix.

**Comment by sklam** [comment]
> What did you have to do on Zenodo side, did it require making an account?

Signup with GH. It automated the webhook creation. I just need to activate it on "numba" (1 mouse click on the Zenodo GUI) and make a release on GH to trigger it.

> Also, where was the list of authors source from, is it automatically derived from git history?

It's automatic. Not sure how though.

**Pull request merged**